storage/inmem/txn.go

```diff
@@ -0,0 +1,410 @@
+// Copyright 2017 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package inmem
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/open-policy-agent/opa/storage"
+)
+
+type transaction struct {
+	xid      uint64
+	write    bool
+	log      *snapshot
```

Comment:
please describe the contents of log: it is a sequence of snapshots, in what order?